src/index.js

```diff
@@ -0,0 +1,26 @@
+'use strict';
+
+const fs = require('fs/promises');
+const { sep, basename } = require('path');
+
+const fileParam = process.argv[2];
+const endpointParam = process.argv[3];
+
+moveFile(fileParam, endpointParam);
+
+async function moveFile(file, endpoint) {
+  try {
+    const dataOfFile = await fs.readFile(file);
+
+    await fs.writeFile(endpoint, dataOfFile);
+    await fs.rm(file);
+  } catch (err) {
+    if (err.code === 'EROFS') {
+      return moveFile(file, `.${endpoint + sep}`);
+    }
+
+    if (err.code === 'EISDIR') {
```

Comment:
The code should throw an error if the endpoint is a directory, instead of attempting to move the file into the directory.

Reply:
Point 4 from example told that we have to move our file to the directory if it exist. Showing error must to be in the case of absent the directory

- `node index file.txt dir/` moves the file under the `dir`. In case `dir` does not exist an error is thrown.